src/uct/ib/ud/base/ud_def.h

```diff
@@ -118,16 +118,22 @@ Control packet header
 */
 
 typedef struct uct_ud_neth {
-    uint32_t            packet_type;
-    uct_ud_psn_t        psn;
-    uct_ud_psn_t        ack_psn;
+    union {
+        struct {
+            uint32_t            packet_type;
+            uct_ud_psn_t        psn;
+            uct_ud_psn_t        ack_psn;
+        };
+        ucs_status_t status; /* used in case of failure */
```

Comment:
readability and safety. I do not like that ud net header may be affected by changes to the ucs_status_t